cmd/kola/spawn.go

```diff
@@ -64,12 +79,28 @@ func runSpawn(cmd *cobra.Command, args []string) {
 func doSpawn(cmd *cobra.Command, args []string) error {
 	var err error
 
+	if spawnDetach {
+		spawnShell = false
+		spawnRemove = false
+		spawnVerbose = true
+		if spawnUserData == "" {
+			spawnSetSSHKeys = true
+		}
+	}
+
 	if spawnNodeCount <= 0 {
 		return fmt.Errorf("Cluster Failed: nodecount must be one or more")
 	}
 
 	var userdata *conf.UserData
-	if spawnUserData != "" {
+	if spawnSetSSHKeys && spawnUserData != "" {
+		return fmt.Errorf("Cannot set SSH keys if userdata is specified")
```

Comment:
It seems like we could support `-k -u...` by merging in the user's ssh keys. Is there a reason not to do that? Just simplicity?

It is true that if `-u` is used it generally should already have ssh keys, but it doesn't seem like they *have* to conflict